pkg/build/webhook/controller.go

```diff
@@ -71,7 +71,11 @@ func (c *controller) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		notFound(w, "Plugin ", uv.plugin, " not found")
 		return
 	}
-	revision, proceed, err := plugin.Extract(buildCfg, uv.secret, uv.path, req)
+	revision, evptr, proceed, err := plugin.Extract(buildCfg, uv.secret, uv.path, req)
+	envvars := []kapi.EnvVar{}
+	if evptr != nil {
+		envvars = *evptr
+	}
```

Comment:
Yeah just cause I've heard / read that ptr params / returns avoid data
copies, where of course that is more desirable for arbitrary sized structs
and arrays.  And in theory the number of build requests could get high with
larger systems.  I don't feel strongly though if you want me to switch.

On Tuesday, April 12, 2016, Ben Parees <notifications@github.com> wrote:

> In pkg/build/webhook/controller.go
> <https://github.com/openshift/origin/pull/8477#discussion_r59457879>:
>
> > @@ -71,7 +71,11 @@ func (c *controller) ServeHTTP(w http.ResponseWriter, req *http.Request) {
> >  		notFound(w, "Plugin ", uv.plugin, " not found")
> >  		return
> >  	}
> > -	revision, proceed, err := plugin.Extract(buildCfg, uv.secret, uv.path, req)
> > +	revision, evptr, proceed, err := plugin.Extract(buildCfg, uv.secret, uv.path, req)
> > +	envvars := []kapi.EnvVar{}
> > +	if evptr != nil {
> > +		envvars = *evptr
> > +	}
>
> seems like you can simplify this and the other spot where you do the same
> thing if you just make Extract return []kapi.EnvVar instead of
> *[]kapi.EnvVar, was there a reason to have it return the pointer?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8477/files/f1b3c665dab68bc1e83ade193a8acd2873f44864#r59457879>
>